community/kernel/src/main/java/org/neo4j/graphdb/traversal/BranchCollisionPolicy.java

```diff
@@ -19,10 +19,37 @@
  */
 package org.neo4j.graphdb.traversal;
 
+import org.neo4j.function.Predicate;
+import org.neo4j.graphdb.Path;
+
 /**
  * Copied from kernel package so that we can hide kernel from the public API.
  */
 public interface BranchCollisionPolicy
 {
+    @Deprecated
     BranchCollisionDetector create( Evaluator evaluator );
+
+    BranchCollisionDetector create( Evaluator evaluator, Predicate<Path> pathPredicate );
+
+    class CollisionPolicyAdapter implements BranchCollisionPolicy
```

Comment:
This is a wrapper, isn't it? What's the reason to have this as a wrapper instead of an adapter (i.e. interface default methods for java < 8)?